packages/compose/src/hooks/use-viewport-match/index.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal dependencies
+ */
+import useMediaQuery from '../use-media-query';
+
+/**
+ * Hash of breakpoint names with pixel width at which it becomes effective.
+ *
+ * @see _breakpoints.scss
+ *
+ * @type {Object}
+ */
+const BREAKPOINTS = {
+	huge: 1440,
+	wide: 1280,
+	large: 960,
+	medium: 782,
+	small: 600,
+	mobile: 480,
+};
+
+/**
+ * Object mapping media query operators to the condition to be used.
+ *
+ * @type {Object}
+ */
+const CONDITIONS = {
+	'>=': 'min-width',
+	'<': 'max-width',
+};
+
+/**
+ * Returns true if the viewport matches the given query, or false otherwise.
+ *
+ * @param {string} query Query string. Includes operator and breakpoint name,
+ *                       space separated. Operator defaults to >=. The supported
+ *                       breakpoint names are: mobile, small, medium, large, wide
+ *                       and huge.
+ *
+ * @example
+ *
+ * ```js
+ * useViewportMatch( '< huge' );
+ * useViewportMatch( 'medium' );
+ * ```
+ *
+ * @return {boolean} Whether viewport matches query.
+ */
+const useViewportMatch = ( query ) => {
+	if ( query.indexOf( ' ' ) === -1 ) {
+		query = '>= ' + query;
+	}
+	const [ operator, breakpoint ] = query.split( ' ' );
+	const mediaQuery = `(${ CONDITIONS[ operator ] }: ${ BREAKPOINTS[ breakpoint ] }px)`;
```

Comment:
I considered whether this might be an opportunity to reconsider the syntax we created for describing operations on the breakpoint sizes, whether we could be more explicit in expressing this logic. I think it's probably fine. The main alternative could just be to have it be passed as an array or two arguments:

```js
useViewportMatch( 'medium' );      // Before: useViewportMatch( 'medium' );
useViewportMatch( '>', 'medium' ); // Before: useViewportMatch( '> medium' );
// Or: (maybe easier to work with as the optional logical comparison argument)
useViewportMatch( 'medium', '>' ); // Before: useViewportMatch( '> medium' );
```

Reply:
We don't really have much precedent for it, but since the syntax here could have a higher likelihood for mistakes, seems like introduce some safety checks and/or logging to deal with an invalid operator or breakpoint being provided.